_release/bat/release_test.py

```diff
@@ -0,0 +1,351 @@
+#
+# Copyright (c) 2016 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import subprocess32
+import os
+import string
+import unittest
+import taprunner
+import time
+import sys
+import random
+
+cliTimeout = 30
+
+class Tenant:
+    def __init__(self, uuid, name):
+        self.uuid = uuid
+        self.name = name
+
+class Workload:
+    def __init__(self, uuid, name, image_uuid, cpus, mem):
+        self.uuid = uuid
+        self.name = name
+        self.image_uuid = image_uuid
+        self.cpus = cpus
+        self.mem = mem
+
+class CNCI:
+    def __init__(self, uuid, tenant_uuid, ip):
+        self.uuid = uuid
+        self.tenant_uuid = tenant_uuid
+        self.ip = ip
+
+class Instance:
+    def __init__(self, uuid, status, ip, mac, cn_uuid, image_uuid, tenant_uuid):
+        self.uuid = uuid
+        self.status = status
+        self.ip = ip
+        self.mac = mac
+        self.cn_uuid = cn_uuid
+        self.image_uuid = image_uuid
+        self.tenant_uuid = tenant_uuid
+
+def ciao_user_env():
+    ciao_env = os.environ.copy()
+    return ciao_env
+
+def ciao_admin_env():
+    ciao_env = os.environ.copy()
+    ciao_env["CIAO_USERNAME"] = ciao_env["CIAO_ADMIN_USERNAME"]
+    ciao_env["CIAO_PASSWORD"] = ciao_env["CIAO_ADMIN_PASSWORD"]
+    return ciao_env
+
+# implement a wait loop that waits for all instances to move to "running"
+# but timeout after so many tries.
+def wait_till_running(uuid):
```

Comment:
@mcastelino yes - you are definitely right there. I've already had to change the number of wait loops since I submitted this PR. I'll make it configurable with a higher default value.

@leoswaldo I think thats a good idea. I'll make a change for that.